internal/auth/auth.go

```diff
@@ -22,66 +24,113 @@ var privToAPI = map[privileges.Privilege]model.Privilege{
 	privileges.VehicleVinCredential:   model.PrivilegeVehicleVinCredential,
 }
 
-// RequiresTokenCheck checks if the tokenID in the context matches the tokenID in the claim.
-func RequiresTokenCheck(ctx context.Context, _ any, next graphql.Resolver) (any, error) {
-	claim, err := getTelemetryClaim(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("%s: %w", errUnauthorized.Error(), err)
-	}
-	fCtx := graphql.GetFieldContext(ctx)
-	if fCtx == nil {
-		return nil, fmt.Errorf("%w: no field context found", errUnauthorized)
-	}
-	tokenID, ok := fCtx.Args["tokenId"].(int)
-	if !ok {
-		return nil, fmt.Errorf("%w: failed to get tokenID from args", errUnauthorized)
+type PrivilegeContractValidator struct {
+	vehicleNFTAddress      common.Address
+	manufacturerNFTAddress common.Address
+	identityService        services.IdentityService
+}
+
+func NewPrivilegeContractValidator(settings config.Settings) *PrivilegeContractValidator {
+	vNFTAddr := common.HexToAddress(settings.VehicleNFTAddress)
+	mNFTAddr := common.HexToAddress(settings.ManufacturerNFTAddress)
+	return &PrivilegeContractValidator{
+		vehicleNFTAddress:      vNFTAddr,
+		manufacturerNFTAddress: mNFTAddr,
+		identityService:        services.NewIdentityService(&settings),
 	}
-	if strconv.Itoa(tokenID) != claim.TokenID {
-		return nil, fmt.Errorf("%w: tokenID mismatch", errUnauthorized)
+}
+
+// VehicleNFTPrivCheck checks if the claim set in the context includes the correct address the required privileges for the VehicleNFT contract.
+func (pcv *PrivilegeContractValidator) VehicleNFTPrivCheck(ctx context.Context, _ any, next graphql.Resolver, privs []model.Privilege) (any, error) {
```

Comment:
I would probably split up the two checkers. They don't share any state and one has this giga dependency on Identity.